client/task-list/task-item.scss

```diff
@@ -1,21 +1,60 @@
 $foreground-color: var(--wp-admin-theme-color);
+$task-alert-yellow: #f0b849;
 
 .woocommerce-task-list__item {
+	position: relative;
+
+	&::before {
+		content: '';
+		position: absolute;
+		top: 0;
+		left: 0;
+		width: 4px;
+		height: 100%;
+		background: transparent;
+	}
+
+	&.is-l3 {
+		&::before {
+			background-color: $alert-red;
+		}
+		.gridicons-notice-outline {
+			fill: $alert-red;
+		}
+	}
+
+	&.is-l2 {
+		&::before {
+			background-color: $task-alert-yellow;
+		}
+	}
+
 	.woocommerce-task-list__item-title {
 		color: $foreground-color;
 	}
 
 	.woocommerce-task-list__item-before {
-		margin-right: 20px;
 		display: flex;
 		align-items: center;
+		padding: $gap 20px $gap $gap-large;
+	}
```

Comment:
As mentioned in the description, I removed the gutters and moved the padding to the individual containers within the task list.